File: src/main/java/hr/tvz/polling/controller/AnketaTemplateController.java
package hr.tvz.polling.controller;



import hr.tvz.polling.service.AnketaTemplateServiceImpl;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.ui.ModelMap;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.ResponseBody;

@Controller
@RequestMapping("/anketaTemplate")
public class AnketaTemplateController {

	@Autowired
	public AnketaTemplateServiceImpl anketaTemplateServiceImpl;
	
	/**
	 * Anketa template
	 */
	@RequestMapping("/anketeList.json")
	public @ResponseBody List<String> getAnketaOptionsList(){
		return anketaTemplateServiceImpl.getOptionsList();
	}
	
    @RequestMapping(value = "/addOption/{option}", method = RequestMethod.POST)
    public @ResponseBody void addAnketaOption(@PathVariable("option") String option) {
        anketaTemplateServiceImpl.addOption(option);
    }
    @RequestMapping(value = "/saveAnketa")//, method = RequestMethod.POST)
    public @ResponseBody void saveAnketa() {	
    	String question = anketaTemplateServiceImpl.getQuestion();
    	List<String> options = anketaTemplateServiceImpl.getOptionsList();
    	int maxVotes = anketaTemplateServiceImpl.getMaxAllowedVoters();
    	anketaTemplateServiceImpl.addMockAnketa(anketaTemplateServiceImpl.createMockAnketa(question, options, maxVotes));
    }
    
    @RequestMapping(value = "/addQuestion/{question}", method = RequestMethod.POST)
    public @ResponseBody void addAnketaQuestion(@PathVariable String question) {
        anketaTemplateServiceImpl.addQuestion(question);
    }
    
    @RequestMapping(value = "/question")//, method = RequestMethod.POST)
    public @ResponseBody String getTemplateQuestion() {
        return anketaTemplateServiceImpl.getQuestion();
    }
    
    
    @RequestMapping(value = "/removeOption/{option}", method = RequestMethod.DELETE)
    public @ResponseBody void removeAnketaOption(@PathVariable String option) {
        anketaTemplateServiceImpl.removeOption(option);
    }
	
    @RequestMapping(value = "/removeAllOptions", method = RequestMethod.DELETE)
    public @ResponseBody void removeAllOptions() {
        anketaTemplateServiceImpl.removeAllOptions();
    }
    
    @RequestMapping(value = "/maxAllowed/{maxVotes}", method = RequestMethod.POST)
    public @ResponseBody void setMaxAllowedVotes(@PathVariable int maxVotes) {
        anketaTemplateServiceImpl.setMaxAllowedVoters(maxVotes);
    }
    
    @RequestMapping("/layout")
    public String getAnketaListPartialPage(ModelMap modelMap) {
        return "anketaTemplate/anketaTemplateLayout";
    }
    
}

File: src/main/webapp/resources/js/controllers/AnketaTemplateController.js
'use strict';

/**
 * AnketaTemplateController
 */
var AnketaTemplateController = function($scope, $http) {
    $scope.fetchAnketaOptionsList = function() {
        $http.get('anketaTemplate/anketeList.json').success(function(anketeList){
            $scope.anketaOptionsList = anketeList;
            $scope.getTemplateQuestion();
            $scope.anketaSaved = false;
            
        });
    };

    $scope.addNewQuestion = function(newQuestion) {
    	$http.post('anketaTemplate/addQuestion/' + newQuestion).success(function() {
// $scope.templatequestion = newQuestion;
    		$scope.questionName = newQuestion;
    		$scope.buttonQuestion = "Change Question";
    		$scope.fetchAnketaOptionsList();
    	});
// $scope.questionName = '';
    };
    
    $scope.getTemplateQuestion = function(){
    	$http.get('anketaTemplate/question').success(function(question){
    		$scope.questionName = question;
    	});
    };
    $scope.addNewOption = function(newOption) {
        $http.post('anketaTemplate/addOption/' + newOption).success(function() {
            $scope.fetchAnketaOptionsList();
        });
        $scope.optionName = '';

    };

    $scope.removeOption = function(option) {
        $http.delete('anketaTemplate/removeOption/' + option).success(function() {
            $scope.fetchAnketaOptionsList();

        });
    };

    $scope.removeAllOptions = function() {
        $http.delete('anketaTemplate/removeAllOptions').success(function() {
            $scope.fetchAnketaOptionsList();

        });

    };
    
    $scope.saveAnketa = function() {
        $http.post('anketaTemplate/saveAnketa/').success(function() {
        	$scope.anketaSaved = true;
        });
    };
    
    $scope.setMaxAllowed = function(maxVotes){
    	$http.post('anketaTemplate/maxAllowed/'+ maxVotes).success(function() {
    		$scope.fetchAnketaOptionsList();
    	});
    };
    
    $scope.fetchAnketaOptionsList();
    $scope.buttonQuestion = "Add question";
};